temporalio/temporal
# Support re-enqueuing DLQ'd replication tasks

**What changed?**
This PR adds support for re-enqueuing replication tasks from the DLQ back to their original queue for queue v2.


**Why?**
Currently, we only support re-enqueuing history tasks for queue v2. Support wasn't added for replication tasks immediately for 2 reasons:
1. Testing it is *much* more complicated
2. It requires the ability to connect to a remote cluster


**How did you test it?**
In addition to unit tests, there's an integration test which uses tdbg merge with queue v1, v2, and replication streaming enabled and disabled. It does this:
1. Execute a workflow with replication tasks that we override to return terminal errors
2. Verify that they all end up in the DLQ using tdbg dlq read
3. Use tdbg dlq merge to re-enqueue them all
4. Verify that the workflow now completes in the standby cluster


**Potential risks**
This won't work if the original source cluster is no longer active, but I'll fix that in a follow-up PR because this isn't running in production yet, and I think that change will probably deserve its own dedicated review anyway.


**Is hotfix candidate?**
